tests/integration/test_integration.py

```diff
@@ -54,15 +56,21 @@ async def test_given_sdcore_bundle_and_gnbsim_deployed_when_start_simulation_the
         if not username or not password:
             raise Exception("NMS credentials not found.")
         configure_sdcore(username, password)
-        juju_helper.juju_wait_for_active_idle(model_name=RAN_MODEL_NAME, timeout=300)
-        action_output = juju_helper.juju_run_action(
-            model_name=RAN_MODEL_NAME,
-            application_name="gnbsim",
-            unit_number=0,
-            action_name="start-simulation",
-            timeout=6 * 60,
-        )
-        assert action_output["success"] == "true"
+        juju_helper.juju_wait_for_active_idle(model_name=RAN_MODEL_NAME, timeout=300, time_idle=30)
+        for _ in range(3):
+            try:
+                action_output = juju_helper.juju_run_action(
+                    model_name=RAN_MODEL_NAME,
+                    application_name="gnbsim",
+                    unit_number=0,
+                    action_name="start-simulation",
+                    timeout=6 * 60,
+                )
+                assert action_output["success"] == "true"
+                return
+            except (AssertionError, KeyError) as e:
+                logger.warning("Error when running simulation: %s. Retrying...", e)
+        assert False
```

Comment:
I'm not sure about this retry loop.Errors happening here are related to our core or gnbsim and should be addressed

Reply:
After discussing with @Gmerold we decided to go ahead with the change, accepting the risk in favor of having more reliable e2e tests that will catch consistent issues.